buttons/button_base.h

```diff
@@ -24,20 +24,29 @@ class ButtonBase : public Looper,
       while (!DebouncedRead()) YIELD();
       prop.Event(button_, EVENT_PRESSED);
       if (millis() - push_millis_ < 500) {
-        prop.Event(button_, EVENT_DOUBLE_CLICK);
+        doubleclicked_ = true;
       } else {
         push_millis_ = millis();
         current_modifiers |= button_;
       }
       while (DebouncedRead()) {
         if (millis() - push_millis_ > 300) {
-          prop.Event(button_, EVENT_HELD);
+          if (!doubleclicked_) {
+            prop.Event(button_, EVENT_HELD);
+          } else {
+            prop.Event(button_, EVENT_DOUBLE_CLICK_HELD);
+            doubleclickedandheld_ = true;
+          }
```

Comment:
Here we handle the three HELD events to ensure that if a double-click occurs, we don't triger the single-click HELD events.  The second bool doubleclickedandheld_ is used to check if, when the button is released, if the DOUBLE_CLICK event should fire.